crates/sui-core/src/authority_server.rs

```diff
@@ -415,6 +414,11 @@ impl ValidatorService {
                 .await
                 .map_err(|e| tonic::Status::internal(e.to_string()))?
         {
+            if consensus_adapter.num_inflight_transactions() > MAX_INFLIGHT_CONSENSUS_TRANSACTIONS {
+                return Err(tonic::Status::resource_exhausted(
```

Comment:
So will this error be returned to the client who submits the certified transaction? (apologies for me not understanding, it's the first time looking into that codepath)

Reply:
If yes I would say we are looking to a different business behaviour here - our system now gets throttled (well for good reason) but user will get a message about this. Just checking if there is anyone else (product?) who might need to become aware of this

Reply:
Currently when the system is under load users will wait and see timeouts. Let me check with sdk folks to see how they handle errors.

Reply:
Switching to reusing `ConsensusListerner` for backpressure instead.